shared/utility/math/filter/ParticleFilter.hpp

```diff
@@ -98,7 +98,7 @@ namespace utility::math::filter {
          */
         ParticleFilter(const StateVec& mean,
                        const StateMat& covariance,
-                       const ResampleMethod& resample_method_ = ResampleMethod::RESIDUAL | ResampleMethod::SYSTEMATIC) {
+                       const int& resample_method_ = ResampleMethod::RESIDUAL | ResampleMethod::SYSTEMATIC) {
```

Comment:
`RESIDUAL | SYSTEMATIC` isn't  an enum element. The default wouldn't bind with clang. If we want to do this differently, let me know.

Reply:
How so, what is the error message? 
Did you try
```suggestion
                       const int& resample_method_ = ResampleMethod(ResampleMethod::RESIDUAL | ResampleMethod::SYSTEMATIC)) {
```

Reply:
Explicitly casting worked. Thanks for the suggestion :+1:

Reply:
Explicit casting is wrong. I do see where c++ is warning about. It's saying that the value that comes out of bitwise oring the two is not an enum. We are using the enum as constant flags rather than as enums. The solution might be to make this enum just constants